src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -11,12 +12,19 @@ internal class EndToEndTraceHelper
         private static string appName;
         private static string slotName;
 
-        // TODO: Replace TraceWriter with structured event client
         private readonly TraceWriter appTraceWriter;
+        private readonly ILogger logger;
+        private readonly bool isAppInsightsLoggingEnabled;
 
-        public EndToEndTraceHelper(TraceWriter appTraceWriter)
+        public EndToEndTraceHelper(JobHostConfiguration config, TraceWriter traceWriter, bool isAppInsightsLoggingEnabled)
         {
-            this.appTraceWriter = appTraceWriter;
+            this.appTraceWriter = traceWriter;
+            this.isAppInsightsLoggingEnabled = isAppInsightsLoggingEnabled;
+
+            if (this.isAppInsightsLoggingEnabled)
+            {
+                this.logger = config.LoggerFactory?.CreateLogger(this.GetType().Namespace);
```

Comment:
We're trying to stick with a naming pattern that starts with `Host.Triggers` for all triggers. So this should be `Host.Triggers.DurableTask`. The filtering allows you to change the filtering level for all categories (i.e. if you want to disable all host logs, you can set up a filter for `"Host" = "None"` and it'll filter anything that starts with `Host`. We should follow that hierarchy here. 

I made a helper that you can use as well: https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/Constants/LogCategories.cs#L46

Reply:
ChrisG was specific about this to be derived based on the namespace to avoid conflicts.  Keeping as ByDesign for now and we can change this down the line if needed.

---
In reply to: [137310517](https://github.com/Azure/azure-functions-durable-extension/pull/37#discussion_r137310517) [](ancestors = 137310517)